test/unit/Chainweb/Test/Pact5/RemotePactTest.hs

```diff
@@ -130,127 +162,131 @@ mkFixture v baseRdb = do
         )
         cancel
 
-    serviceClientEnv <- liftIO $ do
-        let defaultTLSSettings = (HTTP.TLSSettingsSimple True False False TLS.defaultSupported)
-        httpManager <- HTTP.newTlsManagerWith (HTTP.mkManagerSettings defaultTLSSettings Nothing)
-        return $ mkClientEnv httpManager $ BaseUrl
+    let serviceClientEnv = mkClientEnv httpManager $ BaseUrl
             { baseUrlScheme = Https
             , baseUrlHost = "127.0.0.1"
             , baseUrlPort = port
             , baseUrlPath = ""
             }
 
     return $ Fixture
-        { _cutFixture = fixture
+        { _cutFixture = fx
         , _serviceClientEnv = serviceClientEnv
         }
 
+-- generating this cert and making an HTTP manager take quite a while relative
+-- to the rest of the tests, so they're shared globally.
+-- there's no apparent reason to ever switch them out, either.
+cert :: X509CertPem
+key :: X509KeyPem
+(_, cert, key) = unsafePerformIO $ generateLocalhostCertificate @RsaCert 1
+defaultTLSSettings :: HTTP.TLSSettings
+defaultTLSSettings = (HTTP.TLSSettingsSimple True False False TLS.defaultSupported)
+httpManager :: Manager
+httpManager = unsafePerformIO $ HTTP.newTlsManagerWith (HTTP.mkManagerSettings defaultTLSSettings Nothing)
+
 tests :: RocksDb -> TestTree
-tests rdb = testGroup "Pact5 RemotePactTest"
-    [ testCase "pollingInvalidRequestKeyTest" (pollingInvalidRequestKeyTest rdb)
-    , testCase "pollingConfirmationDepthTest" (pollingConfirmationDepthTest rdb)
-    , testCase "spvTest" (spvTest rdb)
-    , testCase "invalidTxsTest" (invalidTxsTest rdb)
-    , testCaseSteps "caplistTest" (caplistTest rdb)
-    ]
-
-pollingInvalidRequestKeyTest :: RocksDb -> IO ()
-pollingInvalidRequestKeyTest baseRdb = runResourceT $ do
+tests rdb = withResource' (evaluate httpManager >> evaluate cert) $ \_ ->
+    testGroup "Pact5 RemotePactTest"
+        [ testCaseSteps "pollingInvalidRequestKeyTest" (pollingInvalidRequestKeyTest rdb)
+        , testCaseSteps "pollingConfirmationDepthTest" (pollingConfirmationDepthTest rdb)
+        , testCaseSteps "pollingMetadataTest" (pollingMetadataTest rdb)
+        -- FIXME: there is an error when the above tests are reordered to after spvTest.
```

Comment:
can you make a ticket detailing this error? or at least just mentioning its existence?

Reply:
Turns out I fixed this. The WithDict stuff was actually buggy, probably due to GHC's context reduction not dealing with it properly. I imagine ImplicitParams has some special stuff.